contrib/python/tests/python/pants_test/contrib/python/checks/tasks/checkstyle/test_checkstyle.py

```diff
@@ -173,7 +173,7 @@ def test_lint_runs_for_blanket_whitelist(self):
                          print(x+7)
                        """))
     target_py3 = self.make_target('a/python:fail3', PythonLibrary, sources=['fail_py3.py'],
-      compatibility=['CPython>=3.6'])
+      compatibility=['CPython>=3.8'])
```

Comment:
If we're just looking for an incompatible interpreter, how about `3.4` which we know we won't support?

Also cc @CMLivingston for his thoughts :)

Reply:
I realized beyond it being better to use something like `3.4`, this solution won't even work. 

For a couple of the tests, as is it will try to execute the linter using Python 3.8, which obviously does not exist. If I were to set it to 3.4 or 3.5, we would have to guarantee the environment has 3.5 available, which while true at the moment for CI is not necessarily true for local development nor when we upgrade Travis to Ubuntu 18.04 in a year or two.

Instead, I've been trying to figure out how to constrain the tests to only allow Python 2 and ignore the `python_setup.py` interpreter constraints. Then the tests will truly be testing the whitelist functionality. I plan to make a dedicated PR for this change, which will block this one.